woocommerce/woocommerce
# Prevent warning if there was no post ID available in some product elements

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





If there was no results in product list block while some product elements were in use, they would throw warning:
- Product Rating
- Product SKU
- Product Sale Badge
- Product Stock Indicator



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to Product Catalog template
2. Remove the content
3. Add Filter by Price block
4. Add Product Collection block
5. Choose Product Catalog collection
6. Inside Product Collection (in Product Template) add following blocks:
  - Product Rating
  - Product SKU
  - Product Sale Badge
  - Product Stock Indicator
 
<img width="356" alt="image" src="https://github.com/woocommerce/woocommerce/assets/20098064/1fc114a6-4e8f-4b00-8756-ced003c93f9b">

7. Save and go to frontend (`/shop`)
8. Change the price filter so there's no results available, for example by choosing really close min and max value:

<img width="548" alt="image" src="https://github.com/woocommerce/woocommerce/assets/20098064/edc629f0-9dae-4022-9d8b-b7b77ee1cc71">

9. Refresh the page
10. **Expected:** There's no warnings in Query Monitor

Before | After
-- | --
<img width="914" alt="image" src="https://github.com/woocommerce/woocommerce/assets/20098064/e068ec46-b1bf-4800-a0d9-886996654ca0"> | ![Screen Shot 2024-03-18 at 15 59 50 PM](https://github.com/woocommerce/woocommerce/assets/20098064/d87fbef2-d0c8-43c3-af5f-34fbc1fd2c25)






### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
